src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -650,6 +650,75 @@ g.test('copy_contents_from_gpu_context_canvas')
     });
   });
 
+g.test('copy_contents_from_ImageBitmapRendering_context_canvas')
+  .desc(
+    `
+  Test HTMLCanvasElement and OffscreenCanvas with ImageBitmapRendering context
+  can be copied to WebGPU texture correctly.
+
+  It creates HTMLCanvasElement/OffscreenCanvas with 'ImageBitmapRendering'.
+  Use fillRect(2d context) to render red rect for top-left,
+  green rect for top-right, blue rect for bottom-left and white for bottom-right on a
+  2d context canvas and create imageBitmap with that canvas. Use transferFromImageBitmap()
+  to render the imageBitmap to source canvas.
+
+  Then call copyExternalImageToTexture() to do a full copy to the 0 mipLevel
+  of dst texture, and read the contents out to compare with the canvas contents.
+
+  Provide premultiplied input if 'premultipliedAlpha' in 'GPUImageCopyTextureTagged'
+  is set to 'true' and unpremultiplied input if it is set to 'false'.
+
+  If 'flipY' in 'GPUImageCopyExternalImage' is set to 'true', copy will ensure the result
+  is flipped.
+
+  The tests covers:
+  - Valid canvas type
+  - Valid ImageBitmapRendering context type
+  - Valid dstColorFormat of copyExternalImageToTexture()
+  - Valid dest alphaMode
+  - Valid 'flipY' config in 'GPUImageCopyExternalImage' (named 'srcDoFlipYDuringCopy' in cases)
+  - TODO(#913): color space tests need to be added
+
+  And the expected results are all passed.
+  `
+  )
+  .params(u =>
+    u
+      .combine('canvasType', kAllCanvasTypes)
+      .combine('dstColorFormat', kValidTextureFormatsForCopyE2T)
+      .combine('dstAlphaMode', kCanvasAlphaModes)
+      .combine('srcDoFlipYDuringCopy', [true, false])
+      .beginSubcases()
+      .combine('width', [1, 2, 4, 15])
+      .combine('height', [1, 2, 4, 15])
+  )
+  .fn(async t => {
+    const { width, height, canvasType, dstAlphaMode } = t.params;
+
+    const canvas = createCanvas(t, canvasType, width, height);
+
+    const imageBitmapRenderingContext = canvas.getContext('bitmaprenderer');
+
+    if (!(imageBitmapRenderingContext instanceof ImageBitmapRenderingContext)) {
+      t.skip(canvasType + ' canvas imageBitmap rendering context not available');
+    }
+
+    const { canvas: sourceContentCanvas, expectedSourceData } = t.init2DCanvasContent({
+      canvasType,
+      width,
+      height,
+    });
+
+    const imageBitmap = await createImageBitmap(sourceContentCanvas, { premultiplyAlpha: 'none' });
+    imageBitmapRenderingContext?.transferFromImageBitmap(imageBitmap);
```

Comment:
Which removes the need for this `?`. (Also note, `!` would have been more appropriate here, because we don't want to skip the call if imageBitmapRenderingContext is null, we want to assert it isn't null.)